tests/cephfs/cephfs_mirroring/test_cephfs_mirroring_with_snap_schedule.py

```diff
@@ -0,0 +1,453 @@
+                )
+                return 1
+
+            log.info("Snap counts and snaps synced match for both subvolumes.")
+            return 0
+        else:
+            log.error("Failed to capture snaps_synced for one or both subvolumes.")
+            return 1
+
+        return 0
+    except Exception as e:
+        log.error(e)
+        log.error(traceback.format_exc())
+        return 1
+    finally:
+        log.info("Clean up the system")
+        log.info("Delete the snapshots")
+        snapshots_to_delete = [
+            f"{kernel_mounting_dir_1}{subvol1_path}.snap/*",
+            f"{fuse_mounting_dir_1}{subvol2_path}.snap/*",
+        ]
+        for snapshot_path in snapshots_to_delete:
+            source_clients[0].exec_command(
+                sudo=True, cmd=f"rmdir {snapshot_path}", check_ec=False
+            )
+
+        snap_util.remove_snap_schedule(
+            source_clients[0], subvol1_path, fs_name="cephfs"
+        ),
+        snap_util.remove_snap_schedule(
+            source_clients[0], subvol2_path, fs_name="cephfs"
+        ),
+
+        log.info("Unmount the paths")
+        paths_to_unmount = [kernel_mounting_dir_1, fuse_mounting_dir_1]
+        for path in paths_to_unmount:
+            source_clients[0].exec_command(sudo=True, cmd=f"umount -l {path}")
+
+        log.info("Remove paths used for mirroring")
+        paths_to_remove = [subvol1_path, subvol2_path]
+        for path in paths_to_remove:
+            fs_mirroring_utils.remove_path_from_mirroring(
+                source_clients[0], source_fs, path
+            )
+
+        log.info("Destroy CephFS Mirroring setup.")
+        fs_mirroring_utils.destroy_cephfs_mirroring(
+            source_fs,
+            source_clients[0],
+            target_fs,
+            target_clients[0],
+            target_user,
+            peer_uuid,
+        )
+
+        log.info("Remove Subvolumes")
+        for subvolume in subvolume_list:
+            fs_util_ceph1.remove_subvolume(
+                source_clients[0],
+                **subvolume,
+            )
+
+        log.info("Remove Subvolume Group")
+        for subvolumegroup in subvolumegroup_list:
+            fs_util_ceph1.remove_subvolumegroup(source_clients[0], **subvolumegroup)
+
+        log.info("Delete the mounted paths")
+        source_clients[0].exec_command(sudo=True, cmd=f"rm -rf {kernel_mounting_dir_1}")
+        source_clients[0].exec_command(sudo=True, cmd=f"rm -rf {fuse_mounting_dir_1}")
+        for client in [source_clients[0], target_clients[0]]:
+            client.exec_command(
+                sudo=True,
+                cmd="ceph config set mon mon_allow_pool_delete true",
+                check_ec=False,
+            )
+        fs_util_ceph1.remove_fs(source_clients[0], source_fs, validate=False)
+        fs_util_ceph1.remove_fs(target_clients[0], target_fs, validate=False)
+
+
+def get_snaps_synced(fs_name, fsid, asok_file, filesystem_id, peer_uuid, path):
```

Comment:
Do you want to move this to mirror_utils file ?

Reply:
We have validate_snap_synchronisation in mirror_utils already which capture other details also.
This one is specific to this test. If any new tests need this. I ll plan to move to mirror_utils.py